### ISSUE TITLE
Partial success in import's part of API

### ISSUE THREAD

**Issue by birutaibm** [opendatabio/opendatabio#215]
In some cases, when importing something, we have a `success` status and some warning messages about skip entries that could not be imported.
It would be interesting if the client could get data sent to this job that was not imported, so the user could correct the issues and retry only with this remaining entries.

**Comment by betovicentini** [comment]
Relates to [](https://github.com/opendatabio/opendatabio-r/issues/24)

**Comment by andrechalom** [comment]
Yes, I think this is very important, and needs to be integrated with the "affected_ids" API (see the issue linked by @betovicentini). If the server simply sent a list of affected ids with NAs marking the skipped entries, would that be enough? Ex, when importing:

> Abarema acreana
Abarema esquisita
Abarema adenophora

If the server answers with `affected_ids = (1617, NA, 1618)` this would mean that the first and third entries are correctly incorporated, but the second encountered an error. 

There is a more fundamental question related to partial success which is: which errors should generate warnings and proceed the import job, and which should raise errors and mark the job as failed? I've attempted to provide some initial guesses at this, but feel free to change this behavior.

**Comment by birutaibm** [comment]
This more fundamental question is exactly what I am thinking about.

I think job should be marked as failed only if it do not import any data (this is not the current behavior, but I think it should be). This way, user can retry the job without problem of to try insert entries that was just inserted at the previous try.

**Comment by andrechalom** [comment]
@betovicentini, what are your thoughts on this? Maybe we should add another marker, "Partial"?

**Comment by birutaibm** [comment]
Going back to this issue, I think their is three types of response for the "affectec_ids":
1. success at import process (add the id to this field)
2. fail to import because the registry was present at odb (I think it will be useful to inform the present id, maybe with some additional notation to show that this is not a new registry)
3. fail to import because of some error at the provided data (I think it will be ok to add NA at this position of "affectec_ids")
Imagine someone try to import 
```
Abarema acreana
Abarema esquisita
```
it returns the affected_ids `1923,NA`. Them someone else try to import
```
Abarema acreana
Abarema esquisita
Abarema adenophora
```
but only the last one is successful imported (with id 2489). I think it would be good if it returns affected_ids `-1923,NA,2489` (the minus sign meaning it is not a new registry). I think it can help to import a chain of data, where the next one depends on the previous imported.

What do you think? Could I implement this solution?

**Comment by andrechalom** [comment]
I'm really conflicted about using negative numbers or other markings. Actually, I think the best result is returning a pair of values, such as

    1923,present
    NA,error
    2489,new

This makes it much more clear to the client, and makes much easier to get aggregate results (such as "The job was finished with 12 new objects, 3 already imported and 0 errors"

**Comment by birutaibm** [comment]
Yes, this looks better than my suggestion!
@betovicentini do you agree with this solution?

**Comment by andrechalom** [comment]
@birutaibm, please proceed. I don't think we should wait for Beto's input.

**Comment by birutaibm** [comment]
Working on that. The affectedIds is currently codified as a json array of integer numbers, this will be changed to a josn array of objects with 2 attributes (status and id). To illustrate, the previous example will be: `[{"id":1923, "status":"present"}, {"id":"NA", "status":"error"}, {"id":2489, "status":"new"}]`

**Comment by andrechalom** [comment]
Working just fine!
